### DISCUSSION TITLE
Conversion of schemas from v2 json to v3 typescript is not clear

### DISCUSSION THREAD

**Discussion by dooferorg** [sanity-io/sanity#3952] [Sanity Studio Support] [answered]
When looking at the documentation for adding schemas, it mentions an 'index.js' file for importing (this apparently changed from the schema.js file). However on a new V3 clean install to a different server/directory, there is an index.ts file instead. I was looking at https://www.sanity.io/docs/migrating-schema-types but it hasn't helped. I made a 'conversion' of a schema to test with, but it does not show up in sanity studio. Sanity studio just says 'no document types'

So I have a few questions..

- How are schemas supposed to be imported from the json files we already have?
- Is there a conversion tool for converting the apparently deprecated JSON schemas to whatever Sanity wants now?
- Is it a 'bad idea' to even go the TypeScript route for me if I am needing to use previously defined JSON schemas?
- Will there be more documentation forthcoming?

**Comment by dooferorg** [comment, marked as answer]
If it helps anyone else, since no help was forthcoming on this,

for their 'book' example, index.ts would need to read:

`import { book} from "./book.ts"
export const schemaTypes = [book]
`

**Comment by OSEH-svg** [comment]
i have same isssue too. and the documentation is not clear on what to do
